docs/monitoring.md

```diff
@@ -0,0 +1,54 @@
+# Network Monitoring
+
+This document describes how to configure network monitoring within the node provided by `aemon` application.
+
+## Configuration
+
+By default monitoring is disabled. To turn it on `monitoring.publisher.pubkey` and `monitoring.active = true` need to be setup. To start posting transaction setup `monitoring.publisher.autostart` to `true`.
+
+To post transactions account's public and private key is needed. For online processing only public key needs to be configured. Offline post-processing can be done by filtering transaction by publisher public key.
+
+## Metrics configuration
+
+Monitoring uses statsd backend provided by [aec_metrics.erl](../apps/aecore/src/aec_metrics.erl). See metrics [configuration](../apps/aeutils/priv/epoch_config_schema.json#L545).
+
+## Metrics
+
+Each metric uses `ae.epoch.aemon.` prefix.
+
+Name                               | Type      | Description
+---------------------------------- | --------- | ---
+`confirmation.delay`               | histogram | Number of keyblock created before signing transaction
+`forks.key`                        | counter   | Count of list keyblocks (always it should be 0)
+`forks.micro`                      | counter   | Count of list microblocks i.e. microforks
+`gen_stats.microblocks.total`      | histogram | Number of microblock in generation
+`gen_stats.tx.monitoring`          | histogram | Number of monitoring transaction in generation
+`gen_stats.tx.total`               | histogram | Number of all transaction in generation
+`publisher.balance`                | gauge     | Publisher balance
+`publisher.post_tx.max_adjustment` | counter   | Transaction posting error:
+`publisher.post_tx.nonce_too_high` | counter   | Transaction posting error:
+`publisher.post_tx.nonce_too_low`  | counter   | Transaction posting error:
+`publisher.post_tx.success`        | counter   | Successful transaction posts
+`publisher.queue.size`             | histogram | Number of transaction posted but not signed on chain
+`publisher.queue.ttl_expired`      | histogram | Number of transaction with expired ttl
+
+## How to read metrics
+
+#### `confirmation.delay`
+represents network latency. A high number might imply a busy network or unfair leaders.
+
+#### `forks.{key,micro}` & `gen_stats.microblocks.total`
+`forks.micro` represents the length of microfork. It shows a minimum number observed by monitoring, not an exact one. By our observation, roughly 33% of microblocks transactions are rewritten to the next generation. Use `gen_stats.microblocks.total` as a reference.
```

Comment:
> By our observation

Can this be rephrased to a more absolute statement. E.g. "Based on behaviour observed in mainnet in the first half of 2019"?

Reply:
updated